nuxeo-management-rest-api/src/test/java/org/nuxeo/rest/management/TestElasticsearchObject.java

```diff
@@ -0,0 +1,156 @@
+public class TestElasticsearchObject extends ManagementBaseTest {
+
+    public static final String GET_DOCUMENTS_QUERY = "SELECT * from Document";
+
+    @Inject
+    protected CoreSession coreSession;
+
+    @Inject
+    protected ElasticSearchService ess;
+
+    @Inject
+    protected ElasticSearchAdmin esa;
+
+    @Test
+    public void shouldRunIndexation() throws Exception {
+        // Init indexes and drop if any
+        esa.initIndexes(true);
+
+        // Initial docs count
+        long initialDocCount = coreSession.query(GET_DOCUMENTS_QUERY).totalSize();
+
+        // Nothing indexed because of disable indexing flag
+        assertEquals(0, ess.query(new NxQueryBuilder(coreSession).nxql(GET_DOCUMENTS_QUERY)).totalSize());
+
+        // Creates a new docs without indexing them
+        createDocuments();
+
+        // Wait for an eventual indexation
+        TransactionHelper.startTransaction();
+        waitForIndexing();
+        assertEquals(0, ess.query(new NxQueryBuilder(coreSession).nxql(GET_DOCUMENTS_QUERY)).totalSize());
+
+        // Starts the ES indexing
+        String path = String.format("site/management/elasticsearch/%s/reindex", coreSession.getRepositoryName());
+        try (CloseableClientResponse response = httpClientRule.post(path, null)) {
+            assertEquals(SC_NO_CONTENT, response.getStatus());
+            waitForIndexing();
+            assertEquals(2 + initialDocCount,
+                    ess.query(new NxQueryBuilder(coreSession).nxql(GET_DOCUMENTS_QUERY)).totalSize());
+        }
+    }
+
+    @Test
+    public void shouldRunFlush() {
+        try (CloseableClientResponse response = httpClientRule.post("site/management/elasticsearch/flush", null)) {
+            assertEquals(SC_NO_CONTENT, response.getStatus());
+        }
+    }
+
+    @Test
+    public void shouldRunFlushOnRepository() {
+        String path = String.format("site/management/elasticsearch/%s/flush", coreSession.getRepositoryName());
+        try (CloseableClientResponse response = httpClientRule.post(path, null)) {
+            assertEquals(SC_NO_CONTENT, response.getStatus());
+        }
+    }
+
+    @Test
+    public void shouldRunOptimize() {
+        try (CloseableClientResponse response = httpClientRule.post("site/management/elasticsearch/optimize", null)) {
+            assertEquals(SC_NO_CONTENT, response.getStatus());
+        }
+    }
+
+    @Test
+    public void shouldRunOptimizeOnRepository() {
+        String path = String.format("site/management/elasticsearch/%s/optimize", coreSession.getRepositoryName());
+        try (CloseableClientResponse response = httpClientRule.post(path, null)) {
+            assertEquals(SC_NO_CONTENT, response.getStatus());
+        }
+    }
+
+    @Test
+    public void shouldFailRunningIndexationWhenRepositoryNotExist() {
+        // Launch the ES indexing
+        String path = String.format("site/management/elasticsearch/%s", "unExistingRepository");
```

Comment:
You're missing the `reindex` part of the path right?

Reply:
yep you right thx,